projects/opendr_ws/src/perception/scripts/audiovisual_emotion_recognition.py

```diff
@@ -116,16 +120,36 @@ def callback(self, image_data, audio_data):
 
 def select_distributed(m, n): return [i*n//m + n//(2*m) for i in range(m)]
 
-if __name__ == '__main__':
-    device = 'cuda' if torch.cuda.is_available() else 'cpu'
 
+if __name__ == '__main__':
     parser = argparse.ArgumentParser()
-    parser.add_argument('--video_topic', type=str, help='listen to video input data on this topic')
-    parser.add_argument('--audio_topic', type=str, help='listen to audio input data on this topic')
-    parser.add_argument('--buffer_size', type=float, default=3.6, help='size of the audio buffer in seconds')
+    parser.add_argument("--input_video_topic", type=str, default="/usb_cam/image_raw",
+                        help="Listen to video input data on this topic")
+    parser.add_argument("--input_audio_topic", type=str, default="/audio/audio",
+                        help="Listen to audio input data on this topic")
+    parser.add_argument("--output_emotions_topic", type=str, default="/output/output_emotions_topic",
+                        help="Topic name for output emotions recognition")
+    parser.add_argument("--buffer_size", type=float, default=3.6,
+                        help="Size of the audio buffer in seconds")
+    parser.add_argument("--device", type=str, default="cuda",
+                        help="Device to use (cpu, cuda)", choices=["cuda", "cpu"])
     args = parser.parse_args()
 
-    avnode = AudiovisualEmotionNode(input_video_topic=args.video_topic, input_audio_topic=args.audio_topic,
-                                    annotations_topic="/opendr/audiovisual_emotion",
+    # Select the device for running
+    try:
+        if args.device == "cuda" and torch.cuda.is_available():
+            device = "cuda"
+        elif args.device == "cuda":
+            print("GPU not found. Using CPU instead.")
+            device = "cpu"
+        else:
+            print("Using CPU")
+            device = "cpu"
+    except:
+        print("Using CPU")
+        device = "cpu"
```

Comment:
```suggestion
    # Select the device for running the
    try:
        if torch.cuda.is_available():
            print("GPU found.")
            device = 'cuda'
        else:
            print("GPU not found. Using CPU instead.")
            device = 'cpu'
    except:
        device = 'cpu'
```

Reply:
I think it might be sufficient to do it like this

Reply:
Your suggestion does not allow users to choose `cpu` when the device has `cuda`.

Reply:
Hey @ad-daniel, the code used by @minhquoc0712 for device selection is the same updated code we use in all other updated nodes. 

This logic needs to use args.device to get the user's selection in case they want to switch devices and cover the case where cuda is selected but is not available, printing a relevant message.